heap/max_heap.py

```diff
@@ -1,21 +1,116 @@
 class Heap:
-  def __init__(self):
-    self.storage = []
+    def __init__(self):
+        self.storage = []
 
-  def insert(self, value):
-    pass
+        # -- Notes --
+        # Max Heap: A[PARENT(i) ] ≥ A[i]
+        # Min Heap: A[PARENT(i) ] ≤ A[i]
 
-  def delete(self):
-    pass
+    # Inserting a new value to the end of the list and then sift through the list accordingly to max heap
+    def insert(self, value):
+        self.storage.append(value)
+        return self._bubble_up(len(self.storage) - 1)
 
-  def get_max(self):
-    pass
+        # -- Notes --
+        # A[A.length] = v
+        # i = A.length-1
+        # while (i > 1 && A[parent(i)] < A[i])
+        # swap(A[i], A[parent(i)])
 
-  def get_size(self):
-    pass
+    """   * `delete` removes and returns the 'topmost' value from the heap; 
+        this method needs to ensure that the heap property is maintained after 
+        the top most element has been removed. """
+    def delete(self):
+        pass
 
-  def _bubble_up(self, index):
-    pass
+        # -- Notes --
+        #  Delete will swap the root node with the last node of the array, delete the last value and start sift stuff down 
 
-  def _sift_down(self, index):
-    pass
+    # Getting the largest number
+    def get_max(self):
+        max_value = self.storage[0]
+        for i in range(len(self.storage)):
+            if self.storage[i] > max_value:
+                max_value = self.storage[i]
+        return max_value
+
+    # * `get_size` returns the number of elements stored in the heap.
+    def get_size(self):
+        print(len(self.storage))
+
+
+
+    # Child Element bigger than the Parent (Insert)
+    def _bubble_up(self, index):
+        # left_child = (index * 2) + 1
+        # right_child = (index * 2) + 2
+        parent = (index - 1) // 2
+        # Base case to break out of recursion 
+        if index == 0 or self.storage[parent] > self.storage[index]:
```

Comment:
looks good, but I'm concerned that this looks like it will only run once, and then stop. what if I need to bubble up more than once?